### PR TITLE
Add double linear search

### PR DESCRIPTION
**Pull request by Markaeus**
#### Description of Change
Added the cpp implementation of the double linear search based on the python variation with tests included.

#### Checklist
- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: search/double_linear_search.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
+#include <iostream>
```
Comment (Panquesito7) [inline search/double_linear_search.cpp:2]: Please describe how the algorithm works, a Wikipedia or any other source for algorithm explanation, what the algorithm does, etc..

File: search/double_linear_search.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+#include <vector>
+#include <numeric>
+#include <cassert>
```
Comment (Panquesito7) [inline search/double_linear_search.cpp]: Please add a one-line description of what the library/header is for (**see the example below**).

```c++
#include <iostream>    /// for IO operations
#include <cassert>    /// for assert
```

File: search/double_linear_search.cpp [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+/ >>> double_linear_search([1, 5, 5, 10], 1)
+/ 0
+/ >>> double_linear_search([1, 5, 5, 10], 5)
+/ 1
+/ >>> double_linear_search([1, 5, 5, 10], 100)
+/ -1
+/ >>> double_linear_search([1, 5, 5, 10], 10)
+/ 3
+/--------------------------------------------------------------------------*/
+int double_linear_search(std::vector<int> list, int search_item) {
```
Comment (Panquesito7) [inline search/double_linear_search.cpp:23]: Functions should be documented explaining the use of its parameters, a brief description, and what it returns.

File: search/double_linear_search.cpp [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        }
+        start_ind += 1;
+        end_ind -= 1;
+    }
+    return -1;
+}
+
+void test(std::vector<int> list, int number, int expected) {
+    auto result = double_linear_search(std::move(list), number);
+    std::cout << "Expected value" << expected << " in search for " << number
```
Comment (Panquesito7) [inline search/double_linear_search.cpp:41]: Functions should always include `assert` checks to make sure it works 100% as excepted (requires the `cassert` library).

File: search/double_linear_search.cpp [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        } else if (list[end_ind] == search_item) {
+            return end_ind;
+        }
+        start_ind += 1;
+        end_ind -= 1;
+    }
+    return -1;
+}
+
+void test(std::vector<int> list, int number, int expected) {
```
Comment (Panquesito7) [inline search/double_linear_search.cpp:39]: Best not to include parameters and include those in the `main` function.
```suggestion
static void test(std::vector<int> list, int number, int expected) {
```

File: search/double_linear_search.cpp [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    auto result = double_linear_search(std::move(list), number);
+    std::cout << "Expected value" << expected << " in search for " << number
+              << ":\t" << (result == expected ? "PASS" : "FAIL") << std::endl;
+}
+
+int main() {
```
Comment (Panquesito7) [inline search/double_linear_search.cpp:45]:
```suggestion
/**
 * @brief Main function
 * @returns 0 on exit
 */
int main() {
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: others/stairs_pattern.cpp [hunk 7]
```diff
@@ -15,7 +15,7 @@ where number of pairs line is given by user
 
 /** main function */
 int main() {
-    int l, st = 2, x, r, z, n, sp;
+    int l = 0, st = 2, x = 0, r = 0, z = 0, n = 0, sp = 0;
     std::cout << "Enter number of pair - ";
```
Comment (krishnacx) [inline others/stairs_pattern.cpp:19]: pairs ?

**Pull request closed without merging**